src/nexusrpc/_common.py

```diff
@@ -48,31 +48,62 @@ def __init__(
         message: str,
         *,
         type: HandlerErrorType,
-        retryable: Optional[bool] = None,
+        retry_behavior: Optional[HandlerErrorRetryBehavior] = None,
     ):
         """
         Initialize a new HandlerError.
 
         :param message: A descriptive message for the error. This will become the
                         `message` in the resulting Nexus Failure object.
 
-        :param type:
+        :param type: The :py:class:`HandlerErrorType` of the error.
 
-        :param retryable:
+        :param retry_behavior: Optional retry behavior for the error.
         """
         super().__init__(message)
         self._type = type
-        self._retryable = retryable
+        self._retry_behavior = retry_behavior
 
     @property
-    def retryable(self) -> Optional[bool]:
+    def retry_behavior(self) -> Optional[HandlerErrorRetryBehavior]:
+        """
+        The retry behavior set for this error.
+        """
+        return self._retry_behavior
+
+    @property
+    def retryable(self) -> bool:
         """
         Whether this error should be retried.
 
-        If None, then the default behavior for the error type should be used.
-        See https://github.com/nexus-rpc/api/blob/main/SPEC.md#predefined-handler-errors
+        If :py:attr:`retry_behavior` is None, then the default behavior for the error
+        type is used. See
+        https://github.com/nexus-rpc/api/blob/main/SPEC.md#predefined-handler-errors
         """
-        return self._retryable
+        if self.retry_behavior == HandlerErrorRetryBehavior.RETRYABLE:
+            return True
+        elif self.retry_behavior == HandlerErrorRetryBehavior.NON_RETRYABLE:
+            return False
+
+        non_retryable_types = {
+            HandlerErrorType.BAD_REQUEST,
+            HandlerErrorType.UNAUTHENTICATED,
+            HandlerErrorType.UNAUTHORIZED,
+            HandlerErrorType.NOT_FOUND,
+            HandlerErrorType.NOT_IMPLEMENTED,
+        }
+        retryable_types = {
+            HandlerErrorType.RESOURCE_EXHAUSTED,
+            HandlerErrorType.INTERNAL,
+            HandlerErrorType.UNAVAILABLE,
+            HandlerErrorType.UPSTREAM_TIMEOUT,
+        }
+        if self._type in non_retryable_types:
+            return False
+        elif self._type in retryable_types:
+            return True
```

Comment:
```suggestion
```

Is there value in having a `retryable_types` set?

Reply:
I notice the other SDKs do it. I think it makes the code clear so I vote keep it (the code says "we expect it to fall into one of these two sets, but we default to `true` if not"

Reply:
:+1: Up to you. I do notice here and in some other places we are creating local sets/dicts inside of every function instead of using globals, but probably not a big deal.